components/url_loader_component/url_loader_component.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_URL_LOADER_COMPONENT_URL_LOADER_COMPONENT_H_
+#define BRAVE_COMPONENTS_URL_LOADER_COMPONENT_URL_LOADER_COMPONENT_H_
+
+#include <list>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "base/callback.h"
+#include "base/memory/weak_ptr.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "url/gurl.h"
+
+namespace network {
+class SharedURLLoaderFactory;
+class SimpleURLLoader;
+}  // namespace network
+
+namespace url_loader_component {
+
+class URLLoaderComponent {
+ public:
+  URLLoaderComponent(
+      net::NetworkTrafficAnnotationTag annotation_tag,
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory);
+  ~URLLoaderComponent();
+
+  using URLRequestCallback =
```

Comment:
URLRequest is very different from URLLoader, so maybe this is just `ResultCallback`

Reply:
Renamed.